daemon/errors.go

```diff
@@ -14,16 +14,12 @@ func (d *Daemon) imageNotExistToErrcode(err error) error {
 			e := fmt.Errorf("No such image: %s", dne.RefOrID)
 			return errors.NewRequestNotFoundError(e)
 		}
-		tag := reference.DefaultTag
-		ref, err := reference.ParseNamed(dne.RefOrID)
+		ref, err := reference.ParseNormalizedNamed(dne.RefOrID)
```

Comment:
Also, it seems like most of this function should be moved to the `Error` method of `ErrImageDoesNotExist`. Then this function would become:

```go
if dne, isDNE := err.(ErrImageDoesNotExist); isDNE {
        return errors.NewRequestNotFoundError(err)
}
return err
```